### DISCUSSION TITLE
Why do I get different results using trainer.predict and trainer.test

### DISCUSSION THREAD

**Discussion by Hayden-CHENPRO** [Lightning-AI/pytorch-lightning#15873] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I got normal results when I used trainer.test, but raise an error about the network using the trainer.predict

here is my code:

	model = MInterface(**self.args)
	model.eval()
	trainer = Trainer(accelerator='auto',devices=1) 

	# model test (get normal results: acc and loss metric)
	trainer.test(model, data_module, ckpt_path=self.model_path)
	# model predict (raise error: File "/share/home/anaconda3/envs/torchlightning/lib/python3.8/site-packages/torch/nn/modules/flatten.py", line 46, in forward return input.flatten(self.start_dim, self.end_dim))
	trainer.predict(model, dataloaders=data_module, ckpt_path=self.model_path)

**Comment by Hayden-CHENPRO** [comment]
other information：

    def test_dataloader(self):
        return DataLoader(self.mnist_test, batch_size=self.batch_size, num_workers=self.num_workers)

    def predict_dataloader(self):
        return DataLoader(self.mnist_predict, batch_size=self.batch_size, num_workers=self.num_workers)
